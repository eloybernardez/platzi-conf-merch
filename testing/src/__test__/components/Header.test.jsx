import React from "react";
import { it, describe, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import ProviderMock from "../../__mocks__/ProviderMock";
import Header from "../../components/Header";

describe("<Header/>", () => {
  it("Render del componente Header", () => {
    render(
      <ProviderMock>
        <Header />
      </ProviderMock>
    );
    expect(screen).toBeDefined();
  });

  it("Render del tÃ­tulo", () => {
    render(
      <ProviderMock>
        <Header />
      </ProviderMock>
    );

    const headerTitle = screen.getByText("Platzi Store");
    expect(headerTitle).toBeDefined();
  });
});

describe("Header Snapshot", () => {
  it("Comprobar el Snapshot de Header", () => {
    const { asFragment } = render(
      <ProviderMock>
        <Header />
      </ProviderMock>
    );
    expect(asFragment(<Header />)).toMatchSnapshot();
  });
});
