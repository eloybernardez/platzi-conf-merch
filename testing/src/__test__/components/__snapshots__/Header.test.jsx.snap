// Vitest Snapshot v1

exports[`Header Snapshot > Comprobar el Snapshot de Header 1`] = `
<DocumentFragment>
  <div
    class="Header"
  >
    <h1
      class="Header-title"
    >
      <a
        href="/"
      >
        Platzi Store
      </a>
    </h1>
    <div
      class="Header-checkout"
    >
      <a
        href="/checkout"
      >
        <i
          class="fas fa-shopping-basket"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
